.github/workflows/test-and-lint.yml

```diff
@@ -96,3 +83,29 @@ jobs:
       working-directory: ./cypress
       run: |
         npm run lint:fix
+  pre-commit:
+    name: Run All Pre-Commit
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.11"
+    - name: Use Node.js 20.x
+      uses: actions/setup-node@v3
+      with:
+        node-version: 20.x
+    - name: Install dependencies Frontend
+      working-directory: ./frontend
+      run: |
+        npm install
+    - name: Install dependencies Backend
+      working-directory: ./backend
+      run: |
+        npm install
```

Comment:
Is this needed? The `backend/` directory doesn't have a `package.json` file - it's just python code. It doesn't look like this action is running `npm install` from the top level directory though (where the CDK code exists). Is that being taken care of elsewhere?

Reply:
My intention was to do it in the root directory. It seems I might not need this at all. Ill try it out